CMakeLists.txt

```diff
@@ -17,6 +17,20 @@ option(RADIUM_WARNINGS_AS_ERRORS    "Treat compiler warning as errors" OFF)
 if ( NOT CMAKE_BUILD_TYPE )
   set( CMAKE_BUILD_TYPE Debug )
 endif()
+
+# guard against in-source builds (source: Eigen)
+if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+  message(FATAL_ERROR "In-source builds not allowed. Please make a new directory (called a build directory) and run CMake from there. You may need to remove CMakeCache.txt. ")
+endif()
+
+# guard against bad build-type strings (source: Eigen)
+string(TOLOWER "${CMAKE_BUILD_TYPE}" cmake_build_type_tolower)
```

Comment:
There is a related piece of code in CompileFlags.cmake (which sets the build type to debug if it is not in the three allowed. )